client/src/debug_config_provider.ts

```diff
@@ -0,0 +1,85 @@
+import {
+  ExtensionContext,
+  debug,
+  WorkspaceFolder,
+  DebugConfiguration,
+  DebugConfigurationProvider,
+  ProviderResult,
+  window,
+} from "vscode";
+
+export interface DenoCmd {
+  name: string;
+  entrypoint: string;
+  security_flags: string;
+  port: string;
+}
+
+export function activeDenoDebug(context: ExtensionContext): void {
+  context.subscriptions.push(
+    debug.registerDebugConfigurationProvider(
+      "deno",
+      new DenoDebugConfigurationProvider()
+    )
+  );
+}
+
+export class DenoDebugConfigurationProvider
+  implements DebugConfigurationProvider {
+  provideDebugConfigurations(): ProviderResult<DebugConfiguration[]> {
+    return [
+      {
+        name: "${2:Debug Deno program}",
+        request: "launch",
+        type: "pwa-node",
+        program: '^"\\${workspaceFolder}/${1:main.ts}"',
+        cwd: '^"\\${workspaceFolder}"',
+        runtimeExecutable: "deno",
+        runtimeArgs: ["run", "--inspect-brk", "${3:-A}"],
+        attachSimplePort: 9229,
+      },
+    ];
+  }
+
+  resolveDebugConfiguration(
+    workspace: WorkspaceFolder | undefined,
+    config: DebugConfiguration
+  ): ProviderResult<DebugConfiguration> {
+    // if launch.json is missing or empty
+    if (!config.type && !config.request && !config.name) {
+      const editor = window.activeTextEditor;
+      const langid = editor?.document.languageId;
+      if (
+        editor &&
+        (langid === "typescript" ||
+          langid === "javascript" ||
+          langid === "typescriptreact" ||
+          langid === "javascriptreact")
+      ) {
+        // https://github.com/microsoft/vscode/issues/106703#issuecomment-694595773
+        // Bypass the bug of the vscode 1.49.0
+        debug.startDebugging(workspace, {
+          name: "Deno: Debug",
+          request: "launch",
+          type: "pwa-node",
+          program: "${file}",
+          runtimeExecutable: "deno",
+          runtimeArgs: ["run", "--inspect-brk", "-A"],
```

Comment:
Deno always defaults to no permissions, so we should do the same in the extension. People can set up permissions themselves in a `launch.json`.

```suggestion
          runtimeArgs: ["run", "--inspect-brk"],
```

Reply:
this one is not for users to change, it's used when you don't have a launch.json and you press f5 and choose deno

Reply:
So the user can't change the permission

Reply:
Yeah I think that is fine. If they need permissions they can add a `/.vscode/launch.json` file. I do think it would make sense to automatically set the `--unstable` flag if `deno.unstable` is set to true.